lib/utils.h

```diff
@@ -98,6 +98,23 @@ inline static const char *getOrdinalText(int i)
     return "th";
 }
 
+inline std::string encodeXml(const std::string& data) 
+{
+    std::string buffer;
+    buffer.reserve(data.size());
+    for(size_t pos = 0; pos != data.size(); ++pos) {
+        switch(data[pos]) {
+            case '&':  buffer.append("&amp;");       break;
+            case '\"': buffer.append("&quot;");      break;
+            case '\'': buffer.append("&apos;");      break;
+            case '<':  buffer.append("&lt;");        break;
+            case '>':  buffer.append("&gt;");        break;
+            default:   buffer.append(&data[pos], 1); break;
+        }
+    }
+    return buffer;
+}
+
```

Comment:
errorlogger has some similar XML encoding code...
Not sure if can easily be used here.

Reply:
here: https://github.com/danmar/cppcheck/blob/aad455e1ea34fa47099ee6b604a1e58bfaf015ea/lib/errorlogger.cpp#L668